grails-app/views/order/_actions.gsp

```diff
@@ -92,9 +92,9 @@
 					</g:link>
 				</div>
 			</g:elseif>
-			<g:isSuperuser>
-				<g:if test="${orderInstance?.orderType == PURCHASE_ORDER}">
-					<g:supports activityCode="${ActivityCode.PLACE_ORDER}">
+			<g:if test="${orderInstance?.orderType == PURCHASE_ORDER}">
+				<g:supports activityCode="${ActivityCode.PLACE_ORDER}">
+					<g:isSuperuser>
```

Comment:
Since this was already there we can leave (and refactor later) but just wanted to point out that this could be a performance issue since _actions.gsp is likely being rendered for each order on the List Orders page. The max results parameter is usually set to 10, so this would be rendered at most 10 times. But it's still not great, especially since it's being called multiple times within the _actions.gsp and each time it's called there are likely database queries involved.

Reply:
With that said, I don't know the best approach here. I think Spring Security uses a thread local to store the authenticated user and their authorities. We do something similar with AuthService but it's not widely used and as far as I can remember I don't think it's being invoked in g:isSuperuser.